src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -1100,37 +1132,227 @@ private void PopulateParamsFromRestRequest(Dictionary<string, object?> parameter
+        /// <param name="fieldNodes">List of ObjectFieldNodes for the the input field.</param>
+        /// <param name="fieldsToAuthorize">Dictionary to store all the entities and their corresponding fields are stored.</param>
+        /// <param name="runtimeConfig">Runtime config.</param>
+        private void ProcessObjectFieldNodesForAuthZ(
+            IMiddlewareContext context,
+            string entityName,
+            InputObjectType schemaObject,
+            IReadOnlyList<ObjectFieldNode> fieldNodes,
+            Dictionary<string, HashSet<string>> fieldsToAuthorize,
+            RuntimeConfig runtimeConfig)
+        {
+            fieldsToAuthorize.TryAdd(entityName, new HashSet<string>());
+            string dataSourceName = GraphQLUtils.GetDataSourceNameFromGraphQLContext(context, runtimeConfig);
+            ISqlMetadataProvider metadataProvider = _sqlMetadataProviderFactory.GetMetadataProvider(dataSourceName);
+            foreach (ObjectFieldNode field in fieldNodes)
             {
-                throw new DataApiBuilderException(
-                    message: "No ClientRoleHeader available to perform authorization.",
-                    statusCode: HttpStatusCode.Unauthorized,
-                    subStatusCode: DataApiBuilderException.SubStatusCodes.AuthorizationCheckFailed);
+                Tuple<IValueNode?, SyntaxKind> fieldDetails = GetFieldDetails(field.Value, context.Variables);
+                SyntaxKind underlyingFieldKind = fieldDetails.Item2;
+
+                // If the SyntaxKind for the field is not ObjectValue and ListValue, it implies we are dealing with a column field
+                // which has an IntValue, FloatValue, StringValue, BooleanValue, NullValue or an EnumValue.
+                // In all of these cases, we do not have to recurse to process fields in the value - which is required for relationship fields.
+                if (underlyingFieldKind is not SyntaxKind.ObjectValue && underlyingFieldKind is not SyntaxKind.ListValue)
+                {
+                    // It might be the case that we are currently processing the fields for a linking input object.
+                    // Linking input objects enable users to provide input for fields belonging to the target entity and the linking entity.
+                    // Hence the backing column for fields belonging to the linking entity will not be present in the source definition of this target entity.
+                    // We need to skip such fields belonging to linking table as we do not perform authorization checks on them.
+                    if (metadataProvider.TryGetBackingColumn(entityName, field.Name.Value, out string? _))
+                    {
+                        // Only add those fields to this entity's set of fields which belong to this entity and not the linking entity.
+                        fieldsToAuthorize[entityName].Add(field.Name.Value);
```

Comment:
do we expect the fields we want to authorize to be mapped names or underlying table's column names?

Reply:
We expect them to be mapped names. The `AuthorizationResolver.AreColumnsAllowedForOperation()` takes care of getting the backing column names:

<img width="584" alt="image" src="https://github.com/Azure/data-api-builder/assets/34566234/aa9c7532-8877-4ecc-91a3-9fd913e2611e">